src/hotspot/share/gc/shared/gcVMOperations.cpp

```diff
@@ -112,7 +112,7 @@ bool VM_GC_Operation::doit_prologue() {
   VM_Heap_Sync_Operation::doit_prologue();
 
   // Check invocations
-  if (skip_operation()) {
+  if (skip_operation() || Universe::heap()->is_shutting_down()) {
```

Comment:
Can the new condition be inlined inside `skip_operation()`?

Reply:
In the first iteration I had it inlined, but then decided to make it more explicit. I can change it back if you prefer